statsmodels/miscmodels/tests/test_generic_mle.py

```diff
@@ -149,3 +150,76 @@ def setup_class(cls):
 
         # Note: loc is fixed, no problems with parameters close to min data
         cls.skip_bsejac = False
+
+
+class TwoPeakLLHNoExog(GenericLikelihoodModel):
+    """Fit height of signal peak over background."""
+    start_params = [10, 1000]
+    cloneattr = ['start_params', 'signal', 'background']
+    exog_names = ['n_signal', 'n_background']
+    endog_names = ['alpha']
+
+    def __init__(self, endog, exog=None, signal=None, background=None,
+                 *args, **kwargs):
+        # assume we know the shape + location of the two components,
+        # so we re-use their PDFs here
+        self.signal = signal
+        self.background = background
+        super(TwoPeakLLHNoExog, self).__init__(endog=endog, exog=exog,
+                                         *args, **kwargs)
+
+    def loglike(self, params):        # pylint: disable=E0202
+        return -self.nloglike(params)
+
+    def nloglike(self, params):
+        endog = self.endog
+        return self.nlnlike(params, endog)
+
+    def nlnlike(self, params, endog):
```

Comment:
Why defining both this and nloglike?

Reply:
small advantage, nlnlike can be used with different endog

but essentially I didn't rewrite the example in the unit test.
I have a partial rewrite in my notebook, for the future.